lib/block_view_lockups.go

```diff
@@ -0,0 +1,1784 @@
+		bav.Snapshot, profilePKID, lockupDurationNanoSecs)
+
+	// Cache the DB entry in the in-memory map.
+	if lockupYieldCurvePoint != nil {
+		bav._setLockupYieldCurvePoint(lockupYieldCurvePoint)
+	}
+
+	return lockupYieldCurvePoint
+}
+
+func (bav *UtxoView) GetLocalYieldCurvePoints(profilePKID *PKID, lockupDuration int64) (
+	_leftLockupPoint *LockupYieldCurvePoint, _rightLockupPoint *LockupYieldCurvePoint) {
+	var leftLockupPoint *LockupYieldCurvePoint
+	var rightLockupPoint *LockupYieldCurvePoint
+
+	// Check the view for yield curve points.
+	if _, pointsInView := bav.PKIDToLockupYieldCurvePointKeyToLockupYieldCurvePoints[*profilePKID]; pointsInView {
+		for _, lockupYieldCurvePoint := range bav.PKIDToLockupYieldCurvePointKeyToLockupYieldCurvePoints[*profilePKID] {
+			// Check for nil pointer cases.
+			if lockupYieldCurvePoint.LockupDurationNanoSecs < lockupDuration && leftLockupPoint == nil {
+				leftLockupPoint = lockupYieldCurvePoint
+			}
+			if lockupYieldCurvePoint.LockupDurationNanoSecs >= lockupDuration && rightLockupPoint == nil {
+				rightLockupPoint = lockupYieldCurvePoint
+			}
+
+			// Check if the point is "more left" than the current left point.
+			if lockupYieldCurvePoint.LockupDurationNanoSecs < lockupDuration &&
+				lockupYieldCurvePoint.LockupDurationNanoSecs > leftLockupPoint.LockupDurationNanoSecs {
+				leftLockupPoint = lockupYieldCurvePoint.Copy()
+			}
+
+			// Check if the point is "more right" than the current right point.
+			if lockupYieldCurvePoint.LockupDurationNanoSecs >= lockupDuration &&
+				lockupYieldCurvePoint.LockupDurationNanoSecs < leftLockupPoint.LockupDurationNanoSecs {
+				rightLockupPoint = lockupYieldCurvePoint.Copy()
+			}
+		}
+	}
+
+	// Now we quickly fetch left and right local yield curve points from the DB using careful seek operations.
+	leftDBLockupPoint, rightDBLockupPoint := DBGetLocalYieldCurvePoints(
+		bav.GetDbAdapter().badgerDb, bav.Snapshot, profilePKID, lockupDuration)
+
+	// Check for nil pointer cases.
+	if leftDBLockupPoint != nil &&
+		leftDBLockupPoint.LockupDurationNanoSecs < lockupDuration {
+		leftLockupPoint = leftDBLockupPoint
+	}
+	if rightDBLockupPoint != nil &&
+		rightDBLockupPoint.LockupDurationNanoSecs >= lockupDuration {
+		rightLockupPoint = rightDBLockupPoint
+	}
+
+	// Check for an updated left and right yield curve point from the DB.
+	if leftDBLockupPoint != nil &&
+		leftDBLockupPoint.ProfilePKID.Eq(profilePKID) &&
+		leftDBLockupPoint.LockupDurationNanoSecs < lockupDuration &&
+		leftDBLockupPoint.LockupDurationNanoSecs > leftLockupPoint.LockupDurationNanoSecs {
+		leftLockupPoint = leftDBLockupPoint
+	}
+	if rightDBLockupPoint != nil &&
+		rightDBLockupPoint.ProfilePKID.Eq(profilePKID) &&
+		rightDBLockupPoint.LockupDurationNanoSecs >= lockupDuration &&
+		rightDBLockupPoint.LockupDurationNanoSecs < rightLockupPoint.LockupDurationNanoSecs {
+		rightLockupPoint = rightDBLockupPoint
+	}
```

Comment:
Whenever you're using something from the db, you generally want to update the view with it. I think here you want to call a _set on the left/right points that you find.

But I think there's another issue I'll put in a secondary comment.